src/index.js

```diff
@@ -221,9 +221,21 @@ class MiniCssExtractPlugin {
 
       compilation.hooks.contentHash.tap(pluginName, (chunk) => {
         const { outputOptions } = compilation;
-        const { hashFunction, hashDigest, hashDigestLength } = outputOptions;
+        const {
+          hashSalt,
+          hashDigest,
+          hashDigestLength,
+          hashFunction,
+        } = outputOptions;
         const hash = createHash(hashFunction);
 
+        if (hashSalt) {
+          hash.update(hashSalt);
+        }
+
+        hash.update(`${chunk.id} `);
```

Comment:
These two are not part of the content of the css file

Reply:
@sokra without it, difference async chunks have same contenthash, but difference content, here example https://github.com/webpack-contrib/mini-css-extract-plugin/issues/454

Reply:
@sokra @evilebottnawi any update on this?

Reply:
As far as I can tell, CSS modules don't have `id` property set so the hash is going to be the same. Would it be better to add `hash.update(m.index2)` (since index2 is used to sort the modules before rendering content asset) below after `m.updateHash(hash)` ?